DigitalLearningSolutions.Web.Tests/Services/GroupServiceTests/GroupsServiceSynchroniseGroupsTests.cs

```diff
@@ -359,6 +370,97 @@ public void
             DelegateMustHaveBeenAddedToGroups(new List<int> { synchronisedGroup1.GroupId, synchronisedGroup2.GroupId });
         }
 
+        [Test]
+        public void UpdateSynchronisedDelegateGroupsBasedOnUserChanges_adds_delegate_to_appropriate_groups()
+        {
+            // Given
+            var centreAnswersData = UserTestHelper.GetDefaultRegistrationFieldAnswers(answer1: "new answer", answer2: "new answer2");
+            A.CallTo(() => clockUtility.UtcNow).Returns(testDate);
+            var synchronisedGroup = GroupTestHelper.GetDefaultGroup(
+                5,
+                "new answer",
+                linkedToField: 1,
+                changesToRegistrationDetailsShouldChangeGroupMembership: true,
+                shouldAddNewRegistrantsToGroup: false
+            );
+            var unsynchronisedGroup = GroupTestHelper.GetDefaultGroup(
+                6,
+                "new answer2",
+                linkedToField: 2,
+                changesToRegistrationDetailsShouldChangeGroupMembership: false,
+                shouldAddNewRegistrantsToGroup: true
+            );
+            A.CallTo(() => groupsDataService.GetGroupsForCentre(centreAnswersData.CentreId))
+                .Returns(new List<Group> { synchronisedGroup, unsynchronisedGroup });
+
+            // When
+            groupsService.UpdateSynchronisedDelegateGroupsBasedOnUserChanges(
+                reusableDelegateDetails.Id,
+                reusableEditAccountDetailsData,
+                centreAnswersData,
+                UserTestHelper.GetDefaultRegistrationFieldAnswers(reusableDelegateDetails.CentreId),
+                null
+            );
+
+            // Then
+            DelegateMustHaveBeenAddedToGroups(new List<int> { synchronisedGroup.GroupId });
+            A.CallTo(
+                () => groupsDataService.AddDelegateToGroup(
+                    A<int>._,
+                    A<int>._,
+                    A<DateTime>._,
+                    A<int>._
+                )
+            ).MustHaveHappenedOnceExactly();

```

Comment:
It's not immediately clear from looking at this test that you are testing both that 
1. the delegate has been added to `synchronisedGroup`, and
2. the delegate has not been added to `unsynchronisedGroup`

Testing that `AddDelegateToGroup` only happened once is a proxy for 2. but it would be better to explicitly test that 
```
 A.CallTo(
    () => groupsDataService.AddDelegateToGroup(
        reusableDelegateDetails.Id,
        unsynchronisedGroup.GroupId,
        A<DateTime>._,
        A<int>._
    )
).MustNotHaveHappened();
````